providers/common/securitycheck.c

```diff
@@ -167,17 +167,25 @@ int ossl_dsa_check_key(OSSL_LIB_CTX *ctx, const DSA *dsa, int sign)
 
         /*
          * For Digital signature verification DSA keys with < 112 bits of
-         * security strength (i.e L < 2048 bits), are still allowed for legacy
-         * use. The bounds given in SP800 131Ar2 - Table 2 are
-         * (512 <= L < 2048 and 160 <= N < 224)
+         * security strength, are still allowed for legacy
+         * use. The bounds given in SP 800-131Ar2 - Table 2 are
+         * (512 <= L < 2048 or 160 <= N < 224).
+         *
+         * We are a little stricter and insist that both minimums are met.
+         * For example a L = 256, N = 160 key *would* be allowed by SP 800-131Ar2
```

Comment:
L=256?

Reply:
I think it would be from a strict reading of the standard which says: `((512 ≤ L < 2048) or (160 ≤ N < 224))`
Satisfying _either_ side makes the _or_ true.  So, 512 ≤ L < 2048 is acceptable regardless of N as is 160 ≤ N < 224 regardless of L.  I think that's stupid so the code is a bit stricter.

Reply:
Looks like that check has been there since at least 1.1.1

Reply:
Yeah, that's relevant too.  Fix pushed.

I suspect that the 131Ar2 standard is being silly.  Rather than spelling out legal cases, it just has a blanket statement that's overly broad.

Reply:
Does this mean you think we should be checking N = 160, 224 and 256 only despite what 131Ar2 says?

Reply:
Maybe OTC decision? I dont mind either way. It has been like this is openSSL for quite some time..

Reply:
Or just leave the fix with what I did first?  That fixes the original issue just fine.
I doubt anyone is broken by the stricter check in `dsa_do_verify()` given how long it's been there.

Reply:
yeah probably - I obviously missed it when doing the check..